dg/lib/error.py

```diff
@@ -0,0 +1,58 @@
+import re
+
+from typing import Union
+
+import colorama
+
+
+class DataGateCLIException(Exception):
+    def __init__(self, error_message: str, error_details: Union[str, None] = None):
+        super().__init__(error_message)
+        self._error_details = error_details
+        self._error_message = error_message
+
+    def __str__(self):
+        output = (
+            f"{colorama.Style.BRIGHT}{self._error_message}{colorama.Style.RESET_ALL}"
+        )
+
+        if self._error_details is not None:
+            output += f" – error details:\n{self._error_details}"
+
+        return output
+
+
+class IBMCloudException(DataGateCLIException):
```

Comment:
We only pass `stderr` to the exception constructor now mostly, as it should be.
I'm wondering though if we can trust the CLI to handle all error output correctly, cleanly writing every information to `stderr`, or if we might be missing out on valuable information, not retaining `stdout` too, in the exception.

I guess this is a case-by-case question. I'm not sure for each exception being re-wrapped now in which `stdout` is being removed from the printed output. So it might be a good idea to retain `stdout` at all places where we're unsure right now and remove them further down the line, if they turn out to be superfluous/empty for a given exception.

So my proposal would be to either offer passing stdout as a optional third parameter of the exception constructor or concatenating it to the detailed error message.

Reply:
Sure, I added a third parameter. If it is passed, an error message would look like this:
```
Error: […]
Standard output:
[…]
```